lib/inspec/cached_fetcher.rb

```diff
@@ -65,5 +68,20 @@ def assert_cache_sanity!
       EOF
       raise exception_message if fetcher.resolved_source[:sha256] != target[:sha256]
     end
+
+    private
+
+    ##
+    # This resolves a bug on compliance whereby uploading new profiles to
+    # automate (and not specifying a version number when running them) causes
+    # old, cached profiles to run. This removes caching when passing a generic
+    # profile name as a target.
```

Comment:
It might be easier to read a comment that's set in the present tense and describes current functionality, instead of needing people reading the code to have to think about bugs that existed in the past.
This comment might just need to describe that sometimes a cache will not be used.